File: apps/mobile-app/ios/VaultModels/EncryptionKeyDerivationParams.swift
import Foundation

public struct EncryptionKeyDerivationParams: Codable {
    public var encryptionType: String?
    public var encryptionSettings: String?
    public var salt: String?

    public init(encryptionType: String? = nil, encryptionSettings: String? = nil, salt: String? = nil) {
        self.encryptionType = encryptionType
        self.encryptionSettings = encryptionSettings
        self.salt = salt
    }
}

File: apps/mobile-app/utils/types/messaging/EncryptionKeyDerivationParams.ts
/**
 * These parameters for deriving encryption key from plain text password. These are stored
 * as metadata in the vault upon initial login, and are used to derive the encryption key
 * from the plain text password in the unlock screen.
 */
export type EncryptionKeyDerivationParams = {
  encryptionType: string,
  encryptionSettings: string,
  salt: string,
};
